crates/kitsune_p2p/kitsune_p2p/src/types/agent_store.rs

```diff
@@ -0,0 +1,121 @@
+//! Data structures to be stored in the agent/peer database.
+
+use crate::types::KitsuneAgent;
+use crate::types::KitsuneP2pError;
+use crate::types::KitsuneSignature;
+use crate::types::KitsuneSpace;
+use url2::Url2;
+
+/// A list of Urls.
+pub type Urls = Vec<Url2>;
+
+/// Value in the peer database that tracks an Agent's representation as signed by that agent.
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
+pub struct AgentInfoSigned {
+    // Raw bytes of agent info signature as kitsune signature.
+    signature: KitsuneSignature,
+    // The agent info.
+    agent_info: AgentInfo,
+}
+
+impl AsRef<KitsuneSignature> for AgentInfoSigned {
+    fn as_ref(&self) -> &KitsuneSignature {
+        &self.signature
+    }
+}
+
+impl AsRef<AgentInfo> for AgentInfoSigned {
+    fn as_ref(&self) -> &AgentInfo {
+        &self.agent_info
+    }
+}
+
+impl AgentInfoSigned {
+    /// Build a new AgentInfoSigned struct given a valid signature of the AgentInfo.
+    // @todo fail this if the signature does not verify against the agent info.
+    // It should not be possible to express a signed agent info type  with no valid signature.
+    pub fn try_new(
+        signature: KitsuneSignature,
+        agent_info: AgentInfo,
+    ) -> Result<Self, KitsuneP2pError> {
+        Ok(Self {
+            signature,
+            agent_info,
+        })
+    }
+
+    /// Thin wrapper around AsRef for KitsuneSignature.
+    pub fn as_signature_ref(&self) -> &KitsuneSignature {
+        self.as_ref()
+    }
+
+    /// Thin wrapper around AsRef for AgentInfo
+    pub fn as_agent_info_ref(&self) -> &AgentInfo {
+        self.as_ref()
+    }
+}
+
+/// Value that an agent signs to represent themselves on the network.
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
+pub struct AgentInfo {
+    // The space this agent info is relevant to.
+    space: KitsuneSpace,
+    // The pub key of the agent id this info is relevant to.
+    id: KitsuneAgent,
```

Comment:
elsewhere in kitsune we use `from_agent` etc rather than `id`

```suggestion
    agent: KitsuneAgent,
```